tests/infrastructure/downward-metrics.go

```diff
@@ -93,4 +99,95 @@ var _ = DescribeInfra("downwardMetrics", func() {
 		Expect(found).To(BeTrue())
 		Expect(metrics.Metrics[j].Value).To(Equal("1"))
 	})
+
+	DescribeTable("should not start a VMI if the feature is disabled until it is re-enabled", func(via libvmi.Option) {
+		By("disabling downwardMetrics feature")
+		tests.DisableDownwardMetrics(virtClient)
+
+		DeferCleanup(func() {
+			By("re-enabling the downwardMetrics feature")
+			tests.EnableDownwardMetrics(virtClient)
+		})
+
+		vmi := libvmifact.NewFedora(via)
+		vmi, err := kubevirt.Client().VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).Create(context.Background(), vmi, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("checking the phase of the VMI")
+		Eventually(matcher.ThisVMI(vmi), 60*time.Second, 5*time.Second).Should(matcher.BeInPhase(v1.Pending))
+
+		By("re-enabling the downwardMetrics feature")
+		tests.EnableDownwardMetrics(virtClient)
+
+		Eventually(matcher.ThisVMI(vmi), 60*time.Second, 5*time.Second).ShouldNot(matcher.BeInPhase(v1.Pending))
+	},
+		Entry("using a disk", libvmi.WithDownwardMetricsVolume("vhostmd")),
+		Entry("using a virtio serial device", libvmi.WithDownwardMetricsChannel()),
+	)
+
+	DescribeTable("should update the VMI conditions if the feature is disabled", func(via libvmi.Option) {
+		vmi := libvmifact.NewFedora(via)
+		vmi = libvmops.RunVMIAndExpectLaunch(vmi, 180)
+		Expect(console.LoginToFedora(vmi)).To(Succeed())
+
+		By("disabling downwardMetrics feature")
+		tests.DisableDownwardMetrics(virtClient)
+
+		DeferCleanup(func() {
+			By("re-enabling the downwardMetrics feature")
+			tests.EnableDownwardMetrics(virtClient)
+		})
+
+		// Triggering an update to force the reconciliation loop. This will avoid to wait much time.
+		Expect(console.LoginToFedora(vmi)).To(Succeed())
+		By("checking the VMI conditions")
+		Eventually(matcher.ThisVMI(vmi), 60*time.Second, 5*time.Second).Should(matcher.HaveConditionFalse(v1.VirtualMachineInstanceOutOfSyncConfiguration))
```

Comment:
You could also asser the reason for this generic condition.

Reply:
Done.